BankClient/Bank.cs

```diff
@@ -2,165 +2,204 @@
 
 public class Bank
 {
-    private void menu()
+    private List<UserInformation> Users = new List<UserInformation>()
     {
-        Console.WriteLine("---------Menu---------");
-        Console.WriteLine("Enter 1 to add client\nEnter 2 to delete client\nEnter 3 to show list of users\nEnter 4 to search one user\nEnter 5 to search group of user\nEnter 6 to change user information");
-        Console.WriteLine("---------Menu ends---------");
-        Console.Write("Choose action: ");
+        new UserInformation("Dmitry", "Lapshin", "1234")
+            {ID = Guid.NewGuid()},
+        new UserInformation("Dima", "Lapshin", "1234")
+            {ID = Guid.NewGuid()}
+    };
+    /// <summary>
+    /// Get information from user
+    /// </summary>
+    /// <returns>Nothing</returns>
+    public UserInformation GetInformationFromUser()
+    {
+        Console.WriteLine("Firstname: ");
+        var firstName = Console.ReadLine();
+        firstName.Trim();
+        Console.WriteLine("Lastname: ");
+        var lastName = Console.ReadLine();
+        lastName.Trim();
+        Console.WriteLine("PhoneNumber: ");
+        var phoneNumber = Console.ReadLine();
+        phoneNumber.Trim();
+        var saveDataToClass = new UserInformation(firstName, lastName, phoneNumber);
+        return saveDataToClass;
+    }
+    /// <summary>
+    /// Function filters list of users which appropriate to inputUser
+    /// </summary>
+    /// <param name="inputUser">Takes information about client which user input</param>
+    /// <returns>List of users which appropriate to input user</returns>
+    public List<UserInformation> SearchUsersInformation(UserInformation inputUser)
+    {
+        var filterList = Users.Where(listUsers =>
+            (listUsers.FirstName == inputUser.FirstName || inputUser.FirstName == "") &&
+            (listUsers.LastName == inputUser.LastName || inputUser.LastName == "") &&
+            (listUsers.PhoneNumber == inputUser.PhoneNumber || inputUser.PhoneNumber == "")).ToList();
+        return filterList;
```

Comment:
This is an extra function and we need to move this logic to `GetAllUsersInformation` function. I suggest you use `string.IsNullOrEmpty() `instead of `inputUser.FirstName == ""`

Reply:
Ok